js/client-battle-tooltips.js

```diff
@@ -37,11 +37,71 @@ var BattleTooltips = (function () {
 	};
 
 	// tooltips
-	BattleTooltips.prototype.tooltipAttrs = function (thing, type, ownHeight) {
-		var roomid = this.room.id;
-		return ' onmouseover="BattleTooltips.showTooltipFor(\'' + roomid + '\', \'' + Tools.escapeHTML('' + thing, true) + '\',\'' + type + '\', this, ' + (ownHeight ? 'true' : 'false') + ')" ontouchstart="BattleTooltips.showTooltipFor(\'' + roomid + '\', \'' + Tools.escapeHTML('' + thing, true) + '\',\'' + type + '\', this, ' + (ownHeight ? 'true' : 'false') + ')" onmouseout="BattleTooltips.hideTooltip()" onmouseup="BattleTooltips.hideTooltip()"';
-	};
+	if (navigator.userAgent.match(/(Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini)/)) {
+		// If user is using a mobile phone or tablet, use touch events
+
+		// Touch delay, pressing finger more than that time will cause the tooltip to open.
+		// Shorter time will cause teh button to click
+		var touchDelayForTap = 300; // ms
+
+		// Each time a finger presses the button this function will be callled
+		// First finger starts the counter, and when last fingers leaves time is checked
+		BattleTooltips._onTouchStart = function (roomid, thing, type, elem, ownHeight) {
+			// Saving number of Touch start happends (number of fingers on teh button,
+		  // though most of the time will be 1, don't won't a nasty bug like tooltip stuck open)
+			// Starts by 1 and increases each time this func is called
+			elem._numOfTaps = elem._numOfTaps !== undefined ? (elem._numOfTaps + 1) : 1;
+
+			// On first tap start counting
+			if (elem._numOfTaps === 1) {
+				setTimeout(function () {
+					// When tap is undefined means `_onTouchEnd` deleted it by the end of the time, perform the action
+					if (elem._numOfTaps === undefined) {
+						// short tap
+						elem._isCalledFromTocuh = true;
+						elem.click();
```

Comment:
@Zarel Actually because of the code here I can't share onclick event on both devices, unless there is another way to call the action for the button right over here.